src/signed-xml.ts

```diff
@@ -299,6 +308,7 @@ export class SignedXml {
     }
 
     if (!this.getReferences().every((ref) => this.validateReference(ref, doc))) {
+      this.signedReferences = []; // reset again
```

Comment:
Why do we have to reset this here? Shouldn't the `.signedReferences` array contain the validated references instead of no references?

If we are going to reset this array, then v7.x should probably just leave `getReferences()` alone and just throw if the references don't validate. In that case, we don't even need a 6.1 release because it would be a no-code-change-just-behavior-change to throw here and leave everything else alone.

I have a branch where I've updated tests to account for `signedReferences` being a different length than `getReferences()`. I want to discuss this before I push that up over what you have.

Reply:
Your point about accessing the signed, authenticated, bytes is valid. Of course, per my previous comment, that isn't what `.signedReferences` contains, but that is another matter.

I want to reemphasize, I'm not against releasing 6.1. I also very much appreciate your work and collaboration. Please keep in mind that this isn't my day job, so I'm doing the best I can to keep code quality high and to help the community here.

My question wasn't about pushing off this change or changing the API, but rather, "why are we resetting this `.signedReferences` here in the case that a single reference fails to validate?" Is there value in leaving `.signedReferences` populated with the references that did validate?

Reply:
Thank you @ahacker1-securesaml . We've always asked for specifications to make sure that we're getting this right and preserving it for the future. In this day and age of people poisoning libraries it is wise for none of us to blindly listen to someone say "I'm awesome, trust me". [Trust, but verify](https://en.wikipedia.org/wiki/Trust,_but_verify).

Thus, in that spirit, I did the research myself and found the specification you must be referring to whereby we've been validating references wrong: https://www.w3.org/TR/xmldsig-core/#sec-CoreValidation

I will update the comments in the code to show this so that we make sure to fix it (breaking change) in v7.x.

Reply:
Thank you for that. In the case that we're dealing with, the data object is a canonicalized XML element, which, due to the rules of canonicalization, is the XML element as a string, encoded to bytes as utf-8. So, it is just super convenient that Node also treats strings as utf-8, so we don't have to work in bytes, we can work in strings, and it all works out. Python 2 would be an entirely different story.